tests/integration/API/PostFile.test.ts

```diff
@@ -0,0 +1,64 @@
+/* tslint:disable:no-relative-imports */
+import { asyncPipe } from 'Helpers/asyncPipe'
+import * as fs from 'fs'
+import * as path from 'path'
+import { prop, identity, head, map } from 'ramda'
+import { describe } from 'riteway'
+
+import { FileHelper } from '../../helpers/files'
+import { IPFS } from '../../helpers/ipfs'
+import { runtimeId, setUpServerAndDb } from '../../helpers/utils'
+
+const PREFIX = `test-functional-node-poet-${runtimeId()}`
+const NODE_PORT = '28081'
+
+const ipfs = IPFS()
+const fileHelper = FileHelper({ port: NODE_PORT  })
+
+const getHash = prop('hash')
+const getStatus = prop('status')
+
+const calcPath = (relPath: string) => path.resolve(__dirname, relPath)
+
+const files = {
+  utf8Markdown: calcPath('../../fixtures/files/utf8.md'),
+}
+
+const awaitAll = (xs: ReadonlyArray<Promise<any>>) => Promise.all(xs)
+
+/* tslint:disable:max-line-length */
+const testMultiPartFileStreamResponseHash = asyncPipe(
+  map(fs.createReadStream),
+  fileHelper.postFileStreams,
+  (x) => x.json(),
+  map(getHash),
+  map(ipfs.fetchFileContent),
+  awaitAll,
+)
+
+const testMultiPartFileStreamResponseStatusCode = asyncPipe(map(fs.createReadStream), fileHelper.postFileStreams, getStatus)
+
+describe('POST /files', async assert => {
+  // Setup Mongodb and the app server
+  const { db, server } = await setUpServerAndDb({ PREFIX, NODE_PORT })
+  {
+    const given = 'a single utf8 mardown file as a multipart file stream'
+
+    assert({
+      given,
+      should: 'return a hash that can be used to verify the file',
+      actual: await testMultiPartFileStreamResponseHash([files.utf8Markdown]).catch(identity),
```

Comment:
Could use some feedback on these complete test functions, not sure if they are better than the normal way of calling the endpoint then testing the response with a bunch of assertions.

Reply:
took me a while to understand what you were doing. It doesn't call out a specific error as readily. It does make it easier to write. What is the `.catch(identity)`? Seems like all you would catch are errors?

Reply:
Yeah its catching any errors so the test does not crash the test runner. Its sort of taking place of riteways `Try` at the moment because riteways `Try` doesn't work for async functions. I've created a pr to adjust `Try` https://github.com/ericelliott/riteway/pull/18

Reply:
> It doesn't call out a specific error as readily.

Yeah its not testing for errors, this assertion is only checking that the returned hashes result in the correct and full file from ipfs. No partial files, etc.

Reply:
I made the tests this way based on your comment above https://github.com/poetapp/node/pull/612/files/0b45faca2c50b636979c9bb9fd64270a551f3fa8#diff-44e42c1f1cb5e71ade2c954f14c5cd3d

I may have taken it too far though